paymentsheet/src/main/java/com/stripe/android/customersheet/CustomerSheetViewModel.kt

```diff
@@ -215,26 +235,32 @@ internal class CustomerSheetViewModel @Inject constructor(
     private fun onItemRemoved(paymentMethod: PaymentMethod) {
         viewModelScope.launch {
             customerAdapter.detachPaymentMethod(paymentMethodId = paymentMethod.id!!)
-                .onFailure {
-                    updateViewState<CustomerSheetViewState.SelectPaymentMethod> {
-                        // TODO (jameswoo) translate error message
-                        it.copy(
-                            errorMessage = "Unable to remove payment method",
-                        )
-                    }
-                }.onSuccess { paymentMethod ->
-                    updateViewState<CustomerSheetViewState.SelectPaymentMethod> { viewState ->
-                        viewState.copy(
-                            savedPaymentMethods = viewState.savedPaymentMethods.filter { pm ->
-                                pm.id != paymentMethod.id!!
-                            },
-                            paymentSelection = viewState.paymentSelection.takeUnless { selection ->
-                                selection is PaymentSelection.Saved &&
-                                    selection.paymentMethod == paymentMethod
-                            },
+                .fold(
```

Comment:
Nit: `fold` seems like the wrong usage here because we’re not returning anything from the expression. We can just do a `when (val result = customerAdapter.detachPaymentMethod(paymentMethodId = paymentMethod.id!!))`.